lib/rubocop/cop/minitest/assert_output.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # This cop checks for opportunities to use `assert_output`.
+      #
+      # @example
+      #   # bad
+      #   $stdout = StringIO.new
+      #   puts object.method
+      #   $stdout.rewind
+      #   assert_match expected, $stdout.read
+      #
+      #   # good
+      #   assert_output(expected) { puts object.method }
+      #
+      class AssertOutput < Cop
+        include MinitestExplorationHelpers
+
+        MSG = 'Use `assert_output` instead of mutating $%<name>s.'
+
+        def on_gvasgn(node)
+          return unless inside_test_case?(node)
+
+          variable_name = node.children[0]
+          name = variable_name[1..-1]
+          return unless name.eql?('stdout') || name.eql?('stderr')
```

Comment:
Can you replace with `Array#include?`?

```diff
--- a/lib/rubocop/cop/minitest/assert_output.rb
+++ b/lib/rubocop/cop/minitest/assert_output.rb
@@ -18,16 +18,16 @@ module RuboCop
       class AssertOutput < Cop
         include MinitestExplorationHelpers

-        MSG = 'Use `assert_output` instead of mutating $%<name>s.'
+        MSG = 'Use `assert_output` instead of mutating %<name>s.'
+        OUTPUT_GLOBAL_VARIABLES = %i[$stdout $stderr].freeze

         def on_gvasgn(node)
           return unless inside_test_case?(node)

-          variable_name = node.children[0]
-          name = variable_name[1..-1]
-          return unless name.eql?('stdout') || name.eql?('stderr')
+          gvar_name = node.children[0]
+          return unless OUTPUT_GLOBAL_VARIABLES.include?(gvar_name)

-          add_offense(node, location: :name, message: format(MSG, name: name))
+          add_offense(node, location: :name, message: format(MSG, name: gvar_name))
         end

         private
```